### DISCUSSION TITLE
Error while downloading certificate

### DISCUSSION THREAD

**Discussion by priyanka-TL** [Sunbird-Lern/Community#19] [Issues] [answered]
@amit-tarento
Like Course Certificate generation, In ML services, We have to generate improvement project certificates. You can refer to the JIRA ticket here for the improvement project certificate story
https://project-sunbird.atlassian.net/browse/SB-28231

For that, I have set up the RC service in my local
https://github.com/Sunbird-RC/demo-certificate-issuance 

docker build was successful and I have created the schema and template using the below reference 
https://project-sunbird.atlassian.net/wiki/spaces/UM/pages/3093921795/Certificate+Schema+Definition

I have generated the certificate and I got the success message 
`{
    "id": "sunbird-rc.registry.create",
    "ver": "1.0",
    "ets": 1656565859785,
    "params": {
        "resmsgid": "",
        "msgid": "9ebe1e1d-c4b7-4cf2-969a-3c9c18eab48c",
        "err": "",
        "status": "SUCCESSFUL",
        "errmsg": ""
    },
    "responseCode": "OK",
    "result": {
        "MeritCertificate": {
            "osid": "1-2022fa79-8fa0-448d-ada7-d5cea1ad5269"
        }
    }
}`

while downloading the certificate as a PDF I'm getting the below error
![Screenshot from 2022-06-30 10-42-11](https://user-images.githubusercontent.com/71248340/176598361-4857c939-ce73-44fb-8171-e155156a6fba.png)

Without passing the application/pdf in the header,  I'm getting a JSON response.

Please help me to fix the error

cc - @aks30 @vijiurs @kiranharidas187 @varsha-sl @Prateek-slokam @aishwaryashikshalokam

**Comment by amitec006** [comment]
@priyanka-TL Please pass below header 
Accept:image/svg+xml

**Reply by priyanka-TL** [reply]
@amit-tarento tried that also same error message

**Reply by priyanka-TL** [reply]
@amit-tarento @reshmi-nair, please help me to fix this

**Reply by priyanka-TL** [reply]
@reshmi-nair Still we are facing an issue with downloading the certificate with the external template. We don't know its an issue with the schema or not. Please help us to figure out the issue

`at dev.sunbirdrc.registry.controller.RegistryEntityController.getEntityType(RegistryEntityController.java:450)
registry_1            | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
registry_1            | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
registry_1            | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
registry_1            | 	at java.lang.reflect.Method.invoke(Method.java:498)
registry_1            | 	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
registry_1            | 	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
`

**Reply by reshmi-nair** [reply]
@priyanka-TL  I hope this issue is resolved. Can you pls update the thread?

**Reply by aks30** [reply]
@reshmi-nair if we can update the confluence doc accordingly so that it will be helpful for someone else in future.

**Comment by priyanka-TL** [comment, marked as answer]
The issue was resolved after modifying the context part of credential_template.json
`"@context": [
    "https://gist.githubusercontent.com/reshmi-nair/386ef6d832236014b7d1037ad4de3da7/raw/8288d9739f3e7946740b4119d0f5f1991695c2a5/v1_context.json",			
    "https://gist.githubusercontent.com/reshmi-nair/d4043c6f74bf98e9fa60232ebf8a6af4/raw/6d17f5bfd48daca5281e5dbef9b384f7cc9d301f/sunbird_context.json"
 ],`

updated context
`"@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://gist.githubusercontent.com/reshmi-nair/386ef6d832236014b7d1037ad4de3da7/raw/8288d9739f3e7946740b4119d0f5f1991695c2a5/v1_context.json",			
    "https://gist.githubusercontent.com/reshmi-nair/d4043c6f74bf98e9fa60232ebf8a6af4/raw/6d17f5bfd48daca5281e5dbef9b384f7cc9d301f/sunbird_context.json"
 ],`
